packages/binary/messagelayer/tangle/tangle.go

```diff
@@ -206,10 +206,14 @@ func (tangle *Tangle) storeMessageWorker(msg *message.Message) {
 
 	// trigger events
 	if tangle.missingMessageStorage.DeleteIfPresent(messageID[:]) {
-		tangle.Events.MissingMessageReceived.Trigger(cachedMessage, cachedMsgMetadata)
+		tangle.Events.MissingMessageReceived.Trigger(&CachedMessageEvent{
+			Message:         cachedMessage,
+			MessageMetadata: cachedMsgMetadata})
 	}
 
-	tangle.Events.MessageAttached.Trigger(cachedMessage, cachedMsgMetadata)
+	tangle.Events.MessageAttached.Trigger(&CachedMessageEvent{
+		Message:         cachedMessage,
+		MessageMetadata: cachedMsgMetadata})
 
 	// check message solidity
 	tangle.solidifierWorkerPool.Submit(func() {
```

Comment:
I can't comment on the actual line.

But `tangle.Events.MessageMissing.Trigger(messageID)` needs to be updated. Please make sure to search the full codebase for `.Trigger(` to check if every event trigger is update.